APIs/src/EpiServer.ContentGraph/Api/Querying/TypeQueryBuilder.cs

```diff
@@ -178,6 +190,92 @@ public TypeQueryBuilder<T> Facet(string propertyName)
                 $"{graphObject.Facets} {facet}";
             return this;
         }
+
+        /// <summary>
+        /// Get facet by field in a nested object
+        /// </summary>
+        /// <typeparam name="TField"></typeparam>
+        /// <param name="fieldSelector">The object</param>
+        /// <param name="propertySelector">Inner property of the object</param>
+        /// <param name="limit">Number of facets to be returned</param>
+        /// <returns></returns>
+        public TypeQueryBuilder<T> Facet<TField>(Expression<Func<T, IEnumerable<TField>>> fieldSelector, Expression<Func<TField, string>> propertySelector, int? limit = null)
```

Comment:
We can put the `limit` option into an option class. I believe we have something like that in here https://github.com/episerver/graph-net-sdk/blob/46ab118b71e0e7049f60f67b3aff8f3ac8a89fde/APIs/src/EpiServer.ContentGraph/Api/Querying/TypeQueryBuilder.cs#L472